### PR TITLE
Add more tests for Query + window bug fix

### PR DESCRIPTION
**Pull request by talIguaz**

### CODE REVIEW

File: pkg/querier/querier.go [hunk 1]
```diff
@@ -65,7 +65,7 @@ func (q *V3ioQuerier) SelectOverlap(name, functions string, step int64, win []in
 func (q *V3ioQuerier) selectQry(name, functions string, step int64, win []int, filter string) (SeriesSet, error) {
 
 	filter = strings.Replace(filter, "__name__", "_name", -1)
-	q.logger.DebugWith("Select query", "func", functions, "step", step, "filter", filter)
+	q.logger.DebugWith("Select query", "func", functions, "step", step, "filter", filter, "win", win)
```
Comment (imakhlin) [inline pkg/querier/querier.go]: win -> window

File: pkg/tsdb/v3iotsdb_integration_test.go [hunk 2]
```diff
@@ -181,6 +183,40 @@ func TestQueryData(t *testing.T) {
...
+			from: 1532940510 + 1, to: 1532940510 + 4,
+			expected: []tsdbtest.DataPoint{}},
+
+		{desc: "Should ingest and query an aggregator", metricName: "cpu",
+			labels: utils.FromStrings("os", "linux", "iguaz", "yesplease"),
+			data: []tsdbtest.DataPoint{{Time: 1532940510, Value: 300.3},
+				{Time: 1532940510 + 5, Value: 300.3},
+				{Time: 1532940510 + 10, Value: 100.4}},
+			from: 1532940510, to: 1532940510 + 11,
+			aggregators: "sum",
```
Comment (imakhlin) [inline pkg/tsdb/v3iotsdb_integration_test.go]: add more than one aggregators test

File: pkg/tsdb/v3iotsdb_integration_test.go [hunk 3]
```diff
@@ -246,19 +282,149 @@ func testQueryDataCase(test *testing.T, v3ioConfig *config.V3ioConfig,
...
-			if !iter.Next() {
-				test.Fatalf("Number of actual data points (%d) is less the expected (%d)", counter, len(data))
-			}
-			t, v := iter.At()
+		actual := iteratorToSlice(iter)
+		assert.ElementsMatch(test, expected, actual)
+		counter++
+	}
+
+	if counter == 0 {
```
Comment (imakhlin) [inline pkg/tsdb/v3iotsdb_integration_test.go]: ok if expected empty result set

File: pkg/tsdb/v3iotsdb_integration_test.go [hunk 4]
```diff
@@ -246,19 +282,149 @@ func testQueryDataCase(test *testing.T, v3ioConfig *config.V3ioConfig,
...
+
+func testQueryDataOverlappingWindowCase(test *testing.T, v3ioConfig *config.V3ioConfig,
+	metricsName string, userLabels []utils.Label, data []tsdbtest.DataPoint, filter string,
+	windows []int, aggregator string,
+	from int64, to int64, expected map[string][]tsdbtest.DataPoint) {
+	defer tsdbtest.SetUp(test, v3ioConfig)()
+
+	var step int64 = 3600
+
+	adapter, err := NewV3ioAdapter(v3ioConfig, nil, nil)
```
Comment (imakhlin) [inline pkg/tsdb/v3iotsdb_integration_test.go]: Move init flow to Setup

File: pkg/tsdb/v3iotsdb_integration_test.go [hunk 5]
```diff
@@ -357,3 +523,12 @@ func TestDeleteTSDB(t *testing.T) {
 		t.Fatal("Did not delete TSDB properly")
 	}
 }
+
+func iteratorToSlice(it chunkenc.Iterator) []tsdbtest.DataPoint {
+	var result []tsdbtest.DataPoint
+	for it.Next() {
+		t, v := it.At()
```
Comment (gtopper) [inline pkg/tsdb/v3iotsdb_integration_test.go]: Shouldn't you be checking `Err()`? `chunkenc.Iterator` doesn't really define a contract for it, but it's there.

File: pkg/tsdb/v3iotsdb_integration_test.go [hunk 6]
```diff
@@ -246,19 +282,149 @@ func testQueryDataCase(test *testing.T, v3ioConfig *config.V3ioConfig,
...
+	}
+
+	set, err := qry.SelectOverlap(metricsName, aggregator, step, windows, filter)
+
+	if err != nil {
+		test.Fatalf("Failed to run Select. reason: %v", err)
+	}
+
+	counter := 0
+	for set.Next() {
```
Comment (gtopper) [inline pkg/tsdb/v3iotsdb_integration_test.go]: Move counter into `for`?

File: pkg/tsdb/v3iotsdb_integration_test.go [hunk 7]
```diff
@@ -181,6 +183,40 @@ func TestQueryData(t *testing.T) {
...
+			aggregators: "sum",
+			expected:    []tsdbtest.DataPoint{{Time: 1532940000, Value: 701.0}}},
+
+		{desc: "Should ingest and query with illegal time (switch from and to)", metricName: "cpu",
+			labels: utils.FromStrings("os", "linux", "iguaz", "yesplease"),
+			data: []tsdbtest.DataPoint{{Time: 1532940510, Value: 314.3},
+				{Time: 1532940510 + 5, Value: 300.3},
+				{Time: 1532940510 + 10, Value: 3234.6}},
+			from: 1532940510 + 1, to: 0,
+			expected: []tsdbtest.DataPoint{}},
```
Comment (gtopper) [inline pkg/tsdb/v3iotsdb_integration_test.go]: If I'm not mistaken, good Go practice is to use `nil` for an empty slice.

**Pull request merged**